### ISSUE TITLE
Nested Contextual Action Panels (CAPs), access from Action button.

### ISSUE THREAD

**Issue by ShikInfor** [infor-design/enterprise-ng#1621]
**Is your feature request related to a problem? Please describe.**
During the development of a widget that involves the integration of multiple Contextual Action Panels (CAPs), a specific challenge emerged when attempting to open a nested CAP from an Action button. The widget architecture included a CAP implemented at the list level and another triggered by an Action button within the main CAP. Following successful development and functionality validation, the team progressed to the implementation phase.

In the course of testing, it became evident that the CAP associated with the Action button failed to open as intended. Instead, the system executed all the necessary API calls without displaying the nested CAP. I also need to confirm that in the text are of CAP we were able to execute and run the CAP.

During the development of a widget that involves the integration of multiple Contextual Action Panels (CAPs), a specific challenge emerged when attempting to open a nested CAP from an Action button. The widget architecture included a CAP implemented at the list level and another triggered by an Action button within the main CAP. Following successful development and functionality validation, the team progressed to the implementation phase.

This anomaly raised concerns about the seamless execution of nested CAPs, specifically when triggered by an Action button. To address this challenge, the team seeks advice on resolving the issue to ensure the effective integration of nested CAPs within the widget's functionality.

**Describe the solution you'd like**
My primary recommendation would be:
1. The First CAP closes to access the second CAP from action button.

**Describe alternatives you've considered**
An alternate solution to the problem statement would be:
1. A tab system to switch between 2 CAP with and addition of back buttons CAP to switch back to the first one

**Comment by tmcconechy** [comment]
@ShikInfor this is all very vague. I think we need an example if you think this a bug? Or do you just want us to make an example so you can see how to do it? 

- make example in https://github.com/infor-design/enterprise
- show two buttons on a page each one opens a different (empty) contextual action panel?
- you have to close the cap and open another one?

We can make the example or you can contribute it? Im not following the idea about:

- A tab system to switch between 2 CAP with and addition of back buttons CAP to switch back to the first one
- The First CAP closes to access the second CAP from action button -> do you mean once you open one cap a button on that cap closes the first CAP and opens another? That should be doable but sounds really clunky from a UI perspective.

Not sure if this needs design @hey-yall ?

**Comment by hey-yall** [comment]
Hi @tmcconechy -- Shikhir submitted this originally on Jira as https://jira.infor.com/browse/IDS-1899 and I have that on hold in case we need design resources. Waiting on more clarity here in the Github ticket before taking action on the design side.

**Comment by tmcconechy** [comment]
@ShikInfor 

1. Did you try to use the CAP api to close the second CAP? I think this should be possible in code.
1. https://github.com/infor-design/enterprise/pull/8348 we will be adding an example with a back button with two "screens" on a CAP that can go back and forth. But that may not solve the case

If you can submit a reduced test case showing the case we can easier work on this.

**Comment by ShikInfor** [comment]
Hello @hey-yall ,
To ease the process for out Business analyst I had developed the IDS ticket and linked the current ticket to the jira.
Hello @tmcconechy ,
I had tested api but couldn't gain access to the second CAP. I am sharing a zip folder with in a test case to access CAP from action button I will refine the code more but for initial tests I hope following attachment will be helpful
[infor.hcm.employee.contextualactionpanel.zip](https://github.com/infor-design/enterprise-ng/files/14120843/infor.hcm.employee.contextualactionpanel.zip)
l

**Comment by tmcconechy** [comment]
@ShikInfor how do we run that? Keep in mind we dont know how to use the widget framework. I was expecting a more isolated example that you can just run in the browser?

**Comment by ShikInfor** [comment]
Hello @tmcconechy, I have tried working with stackblitz.com/ to generate and found that Component have been updated and the set examples aren't working due to component being redefined . Would it be possible to give another means to create an isolated scenario.
Thank You

**Comment by tmcconechy** [comment]
Maybe you could clone https://github.com/infor-design/enterprise-ng-quickstart and put it in there and zip it up.
Or clone https://github.com/infor-design/enterprise-ng and update the examples.

Just any way you can show it running and send the steps and details.

**Comment by ankitsharmaInfor** [comment]
Hi @tmcconechy/@ShikInfor, 
What's the update on this dependency?

**Comment by tmcconechy** [comment]
@ankitsharmaInfor check all the above comments. We are having a very difficult time getting this issue to a place where we can reproduce the issue.

Can you add an example that clearly shows the issue and steps to reproduce? (see other notes to the above).

Currently the issue is holding wait for that information and not being worked on.

**Comment by ankitsharmaInfor** [comment]
@tmcconechy Thanks for the update. @ShikInfor As discussed, please provide the steps to reproduce the issue to Tim so we can fix this.
